### PR TITLE
Update the default CODEOWNERS to point to a new `reviewers` team

### PR DESCRIPTION
**Pull request by freben**
## Summary

Sets the fallback reviewers for pull requests to be a new `reviewers` team, which is a superset of `maintainers`.

Typical additions to that team would be Spotify members that are not full maintainers, and external contributors that are particularly close to the core development. The team will not be very large, but somewhat larger than the list of maintainers.

## Background

The Backstage project sees a very large amount of activity. The project configuration currently has the `maintainers` team set as the gatekeepers of all pull requests, no matter how small or large. This puts pressure on the individual maintainers and can lead to longer lead times than desired.

As Spotify staffs up, and as external contributors move closer to the core of development, I feel that it would be useful to have an intermediate step between "general contributor" and "full maintainer".

I propose that the set of people that shall have the rights to approve/merge pull requests is larger than just the maintainers. This would let us reduce pressure on individual maintainers and to expand the team at Spotify in an effective way. We could more easily perform goalie (rotating designated support person) duties with an extended team, and broaden the review capability to the community.

It is important to note that the reviewers are not expected to behave as maintainers. Issues that are contested, or changes that have significant architectural effects on the code base, shall still be decided upon by maintainers. There is a certain amount of trust involved that these rules are followed. This pull request puts the reviewers team at the root of the entire code base, but future tweaks may put a maintainers team override as an owner of certain sub-directories if desired, to limit the reviewers' area of influence.

### CODE REVIEW

**Comment by iamEAP** [comment]
I've been noodling on this a little bit as well.

It's not a perfect mapping, but the way the Drupal Community handles this is through the concept of marking patches (or PRs) as `RTBC` (_Reviewed and Tested by the Community_).  [Succinct explanation on that here](https://www.bryanbraun.com/2013/01/12/what-does-rtbc-mean/).

If we had such a label, maintainers could focus review efforts to just those PRs marked `RTBC`, empowering the wider community to help flag when community-driven efforts are ready.

Would avoid having to name/manage a `reviewers` group.  ...But would introduce a slightly unique element to the culture of contribution.

**Comment by freben** [comment]
@dzolotusky for visibility

**Review by benjdlambert — APPROVED** [review]

**Comment by freben** [comment]
Converted to draft to more clearly convey that this is just a suggestion open for discussion

**Comment by dzolotusky** [comment]
We should probably add:
1. Clarification on the full responsibilities of a reviewer. For example, I assume reviewers don't have voting rights but that's not stated anywhere in here.
2. Something about a process for becoming a reviewer, for leaving the role, and for where the official list of reviewers will live.
Other than that, I think that I'm a 👍

**Comment by freben** [comment]
Correct, no voting rights. This is basically the right to approve and merge PRs (goalie function), which requires some level of trust to not overstep and merge things that really a maintainer should have a say on first. I had a formulation about not consuming voting slots but softened the wording around it a bit.

Agreed, it should be documented what it means at greater detail. I'll look into writing something about that.

**Comment by freben** [comment]
@dzolotusky Updated the docs a bit.

File: GOVERNANCE.md
Comment (dzolotusky) [inline GOVERNANCE.md:53]: I'm not sure what exactly this means. Does "conference of the maintainers and sponsors" mean that we need a majority of those groups to vote on a reviewer? Does it just mean that having 2 agree maintainers can add or remove any reviewers?

Comment (freben) [reply GOVERNANCE.md:53]: I'd actually appreciate suggestions here, what you see is the result of me struggling a bit to formulate it :) I would like for the end result be that it's really easy for the maintainers to for example add and remove Spotify employees that pitch in but haven't gotten maintainer status.

Comment (dzolotusky) [reply GOVERNANCE.md:53]: I think that we should have a specific definition for what the process is, but try to keep it light. At the moment, I'd suggest "a simple majority vote by the maintainers".
This vote would just be on the PR. So, someone creates a PR adding a person to the reviewers list and once it has a 👍  from a simple majority of maintainers, it's merged.

Comment (HelenGreul) [reply GOVERNANCE.md:53]: Feedback from legal: The main thing to be clear on is the process for adding or removing reviewers (which looks like it was called out already), as well as what constitutes a PR with "significant impact" that requires maintainer approval

Comment (freben) [reply GOVERNANCE.md:53]: Yep - I have been slow in updating this PR but it's on my todo

Comment (freben) [reply GOVERNANCE.md]: Updated! Let me know if there's something that needs adjusting here

**Review by OrkoHunter — APPROVED** [review]
:shipit:

**Review by austinlamon — APPROVED** [review]

**Comment by HelenGreul** [comment]
Ship it! 👍

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**
